src/nvim/os/rstream.c

```diff
@@ -168,7 +168,11 @@ void rstream_stop(RStream *rstream)
 static void alloc_cb(uv_handle_t *handle, size_t suggested, uv_buf_t *buf)
 {
   RStream *rstream = handle_get_rstream(handle);
+  // If we are at the end of the ring buffer, a (long) input sequence
+  // could be read in two steps and be misinterpreted.
+  rbuffer_reset(rstream->buffer);
   buf->base = rbuffer_write_ptr(rstream->buffer, &buf->len);
+  assert(buf->len >= 6);
```

Comment:
@fwalch, I removed the modifications to `alloc_cb()` and placed the call to `rbuffer_reset()` at the end of `read_cb()`. With this we can be sure the next call to `alloc_cb()` allocates the complete ring buffer. No need for an `assert()` anymore.